### PR TITLE
Added `VaultsTable` and stories

### LINKED ISSUE #434: Create `VaultLiquidationsTable` component
### Goal
Modify the current AuctionsTable to function for our new VaultsUI.

### Assets
Screenshot 2022-08-30 at 13 48 10

### PR DESCRIPTION
**Pull request by zoey-kaiser**
Closes #434

Screenshot 2022-09-12 at 16 10 20

Checklist:
- [X] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [X] issue checkboxes are all addressed
- [X] manually checked my feature / not applicable
- [X] wrote tests / not applicable
- [X] attached screenshots / not applicable

### CODE REVIEW

**Comment by zoey-kaiser** [comment]
> General comment: Shouldn't the `vault/` directory be placed under `~/components/auction/`? Here, it's placed under `~/components/` directly.

A vault is not a auction yet. Therefore I think it should not be in the auction directory. What do you think @valiafetisov?

File: frontend/components/vault/VaultsTable.vue [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+                },
+                {
+                    title: 'Collateral amount',
+                    dataIndex: 'collateralAmount',
+                    scopedSlots: { customRender: 'collateralAmount' },
+                    sorter: compareBy('collateralAmount'),
+                    filters: currenciesFilters,
+                    onFilter: (selectedCurrency: string, auction: Auction) => {
+                        return auction.collateralSymbol.includes(selectedCurrency);
+                    },
```
[the remark this answers is not in the document]
Comment (zoey-kaiser) [reply frontend/components/vault/VaultsTable.vue:159]: I am surprised our linting and testing did not catch this @valiafetisov

**Comment by valiafetisov** [comment]
> A vault is not a auction yet. Therefore I think it should not be in the auction directory

Agree, let's put vault-related components into a separate folder

File: frontend/helpers/generateFakeVault.ts [hunk 2]
```diff
@@ -148,8 +149,11 @@ export const generateFakeVaultTransactions = function (
...
 ) {
-    const vaults = [];
-    vaults.push(Array(liquidatedVaultsAmount).fill(null).map(generateFakeVaultLiquidatedTransaction));
-    vaults.push(Array(notLiquidatedVaultsAmount).fill(null).map(generateFakeVaultNotLiquidatedTransaction));
+    let vaults: VaultTransaction[] = [];
+    vaults = [...Array(liquidatedVaultsAmount).fill(null).map(generateFakeVaultLiquidatedTransaction)];
+    vaults = [
+        ...vaults,
+        ...Array(notLiquidatedVaultsAmount).fill(null).map(generateFakeVaultNotLiquidatedTransaction),
+    ];
```
Comment (valiafetisov) [inline frontend/helpers/generateFakeVault.ts:157]:
```suggestion
    const vaults: VaultTransaction[] = [
        ...Array(liquidatedVaultsAmount).fill(null).map(generateFakeVaultLiquidatedTransaction),
        ...Array(notLiquidatedVaultsAmount).fill(null).map(generateFakeVaultNotLiquidatedTransaction),
    ];
```
- [ ] Could you please also shuffle it via `faker.helpers.shuffle()` to check the sorting function?

Comment (zoey-kaiser) [reply frontend/helpers/generateFakeVault.ts:157]: Added it and the sorting method still works!

Comment (valiafetisov) [reply frontend/helpers/generateFakeVault.ts:157]: Why didn't you accept the suggestion? 🙂

File: frontend/components/vault/VaultsTable.vue [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                <format-currency
+                    v-if="collateralAmount && record.collateralType"
+                    :value="collateralAmount"
+                    :currency="record.collateralType"
+                />
+                <span v-else class="opacity-50">Unknown</span>
+            </div>
+            <div slot="proximityToLiquidation" slot-scope="proximityToLiquidation">
+                <template v-if="proximityToLiquidation">
+                    <format-percentage :value="proximityToLiquidation" />
```
Comment (valiafetisov) [inline frontend/components/vault/VaultsTable.vue:25]: Didn't you use `AnimatedNumber` to format the same number in the `Vault` table? I think they would be then displayed differently, right?

File: frontend/helpers/generateFakeVault.ts (hunk 2 above)
Comment (zoey-kaiser) [reply frontend/helpers/generateFakeVault.ts:157]: I missed it! Ill add it now.

**Review by valiafetisov — APPROVED** [review]

File: frontend/components/vault/VaultsTable.stories.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    .add('Fetching With Vaults', () => ({
+        ...common,
+        template:
+            '<VaultsTable :vaultTransactions="vaultTransactions" :last-updated="lastUpdated" :is-loading="true" />',
+    }))
+    .add('Fetching without Vaults', () => ({
+        ...common,
+        template: '<VaultsTable :last-updated="lastUpdated" :is-loading="true" />',
+    }))
+    .add('Empty auctions', () => ({
```
Comment (LukSteib) [inline frontend/components/vault/VaultsTable.stories.js:33]: let's call the story `empty vaults` instead to prevent confusion.

**Review by LukSteib — CHANGES_REQUESTED** [review]

**Review by LukSteib — APPROVED** [review]

**Pull request merged**